src/utils/VerifiableCredentialService.ts

```diff
@@ -0,0 +1,137 @@
+import { AppError } from "./AppError";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { Constants } from "./Constants";
+import { randomUUID } from "crypto";
+import { CopCheckResult } from "../models/enums/CopCheckResult";
+import { MessageCodes } from "../models/enums/MessageCodes";
+
+export class VerifiableCredentialService {
+	readonly tableName: string;
+
+	readonly logger: Logger;
+
+	readonly issuer: string;
+
+	private readonly kmsJwtAdapter: KmsJwtAdapter;
+
+	private static instance: VerifiableCredentialService;
+
+	constructor(tableName: any, kmsJwtAdapter: KmsJwtAdapter, issuer: any, logger: Logger) {
+		this.issuer = issuer;
+		this.tableName = tableName;
+		this.logger = logger;
+		this.kmsJwtAdapter = kmsJwtAdapter;
+	}
+
+	static getInstance(tableName: string, kmsJwtAdapter: KmsJwtAdapter, issuer: string, logger: Logger): VerifiableCredentialService {
+		if (!VerifiableCredentialService.instance) {
+			VerifiableCredentialService.instance = new VerifiableCredentialService(tableName, kmsJwtAdapter, issuer, logger);
+		}
+		return VerifiableCredentialService.instance;
+	}
+
+	getSuccessEvidenceBlock(journeyId: string): VerifiedCredentialEvidence {
+		return {
+			type: Constants.IDENTITY_CHECK,
+			txn: journeyId,
+			strengthScore: 3,
+			validityScore: 2,
+			checkDetails: [
+				{
+					checkMethod: "data",
+					identityCheckPolicy: "none",
+				},
+			],
+		};
+	}
+
+	getFailureEvidenceBlock(journeyId: string): VerifiedCredentialEvidence {
+		return {
+			type: Constants.IDENTITY_CHECK,
+			txn: journeyId,
+			strengthScore: 3,
+			validityScore: 0,
+			failedCheckDetails: [
+				{
+					checkMethod: "data",
+					identityCheckPolicy: "none",
+				},
+			],
+			ci: [
+				"D15",
+			],
+		};
+	}
+	
+
+	async generateSignedVerifiableCredentialJwt(sessionItem: ISessionItem, nameParts: PersonIdentityNamePart[], bankAccountInfo: BankAccountInfo, getNow: () => number): Promise<string> {
+		const now = getNow();
+		const subject = sessionItem.subject;
+		let evidenceInfo;
+		evidenceInfo = sessionItem.copCheckResult === CopCheckResult.FULL_MATCH ?
+			this.getSuccessEvidenceBlock(sessionItem.clientSessionId) : this.getFailureEvidenceBlock(sessionItem.clientSessionId);
+		const verifiedCredential: VerifiedCredential = new VerifiableCredentialBuilder(nameParts, bankAccountInfo, evidenceInfo)
+			.build();
+		const result = {
+			sub: subject,
+			nbf: now,
+			iss: this.issuer,
+			iat: now,
+			jti: randomUUID(),
+			vc: verifiedCredential,
+		};
+
+		this.logger.info("Generated VerifiableCredential jwt", {
+    		jti: result.jti,
+    	});
+		try {
+			// Sign the VC
+			return await this.kmsJwtAdapter.sign(result);
+		} catch (error) {
+			this.logger.error("Error generating signed verifiable credential jwt", {
+				error,
+				messageCode: MessageCodes.ERROR_SIGNING_VC,
+			});
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Server Error");
+		}
+	}
+}
+
+class VerifiableCredentialBuilder {
```

Comment:
This class just seems to set and return an object, is there a reason it can't just be a function like `getSuccessEvidenceBlock` and `getFailureEvidenceBlock`?

Reply:
Figured it means in the future if we wanna add more things to the VC it'd be updating the obj returned in the class rather than updating individual functions?